### PR TITLE
Renaming

### PR DESCRIPTION
**Pull request by Thoralf-M**
Rename:
Message to Block, 
FeatureBlock to Feature, 
UnlockBlock to Unlock, 
parentMessageIds to parents, 
tipsMessageIds to tips

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

Tests, nodejs binding still fails, because the @iota/types package needs to be updated first

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -15,9 +15,9 @@ categories = [ "cryptography::cryptocurrencies" ]
 normal = [ "async-trait", "derive_builder", "slog-stdlog" ]
 
 [dependencies]
-bee-rest-api = { git = "https://github.com/iotaledger/bee", rev = "b64862a1f9a3d66582d34a8462668d80b3296588", default-features = false }
-bee-message = { git = "https://github.com/iotaledger/bee", rev = "b64862a1f9a3d66582d34a8462668d80b3296588", default-features = false, features = [ "serde", "dto" ] }
-bee-pow = { git = "https://github.com/iotaledger/bee", rev = "b64862a1f9a3d66582d34a8462668d80b3296588", default-features = false }
+bee-rest-api = { git = "https://github.com/iotaledger/bee", rev = "146ce1011fd76f3d9f07390b002b29dd6762daf3", default-features = false }
+bee-block = { git = "https://github.com/iotaledger/bee", rev = "146ce1011fd76f3d9f07390b002b29dd6762daf3", default-features = false, features = [ "serde", "dto" ] }
```
Comment (thibault-martinez) [inline Cargo.toml:18]: Order!

File: bindings/nodejs/test/fixtures/signedTransaction.json [hunk 2]
```diff
@@ -40,7 +40,7 @@
             }
         ]
     },
-    "unlockBlocks": [
+    "blocks": [
```
Comment (thibault-martinez) [inline bindings/nodejs/test/fixtures/signedTransaction.json]: Is this correct ?

File: bindings/nodejs/examples/offline_signing/example_signed_transaction.json [hunk 3]
```diff
@@ -40,7 +40,7 @@
             }
         ]
     },
-    "unlockBlocks": [
+    "blocks": [
```
Comment (thibault-martinez) [inline bindings/nodejs/examples/offline_signing/example_signed_transaction.json]: Is this correct ?

File: bindings/python/native/iota_client/client.py [hunk 4]
```diff
@@ -13,7 +13,7 @@ def __init__(self, client_config=None):
         if client_config:
             client_config = dumps(client_config)
 
-        # Create the message handler
+        # Create the block handler
```
Comment (thibault-martinez) [inline bindings/python/native/iota_client/client.py]: Revert

File: examples/08_data_block.rs [hunk 5]
```diff
@@ -16,22 +16,19 @@ async fn main() -> Result<()> {
...
         .with_tag("Hello")
         .with_data("Tangle".as_bytes().to_vec())
         .finish()
         .await?;
 
-    println!(
-        "Message sent https://explorer.iota.org/devnet/message/{}\n",
-        message.id()
-    );
+    println!("Block sent https://explorer.iota.org/devnet/block/{}\n", block.id());
```
Comment (thibault-martinez) [inline examples/08_data_block.rs:26]: Is this URL correct ?

File: examples/09_transaction.rs [hunk 6]
```diff
@@ -41,8 +41,8 @@ async fn main() -> Result<()> {
         .await?;
 
     println!(
-        "Transaction sent: https://explorer.iota.org/devnet/message/{}",
-        message.id()
+        "Transaction sent: https://explorer.iota.org/devnet/block/{}",
```
Comment (thibault-martinez) [inline examples/09_transaction.rs:44]: Correct ?

File: documentation/docs/specs.md [hunk 7]
```diff
@@ -766,7 +766,7 @@ struct SignatureLockedSingleOutput {
     amount: u64,
 }
 
-enum UnlockBlock {
+enum Block {
```
Comment (thibault-martinez) [inline documentation/docs/specs.md]: 🤔

File: documentation/docs/specs.md [hunk 8]
```diff
@@ -729,7 +729,7 @@ enum Payload {
 
 struct Transaction {
     pub essence: TransactionPayloadEssence,
-    pub unlock_blocks: Vec<UnlockBlock/>,
+    pub blocks: Vec<Block/>,
```
Comment (thibault-martinez) [inline documentation/docs/specs.md]: 🤔

File: documentation/docs/libraries/nodejs/api_reference.md [hunk 9]
```diff
@@ -943,7 +943,7 @@ All fields are optional.
 | Field         | Type                        | Description         |
 | ------------- | --------------------------- | ------------------- |
 | essence       | `TransactionPayloadEssence` | Transaction essence |
-| unlock_blocks | `UnlockBlock[]`             | Unlock blocks       |
+| blocks | `Block[]`             | Unlock blocks       |
```
Comment (thibault-martinez) [inline documentation/docs/libraries/nodejs/api_reference.md:946]: :trollface:

File: documentation/docs/libraries/nodejs/api_reference.md [hunk 10]
```diff
@@ -967,14 +967,14 @@ All fields are optional.
 | type  | `SignatureLockedSingle`             | Output type identifier |
 | data  | `{ address: string, amount: number }` | The output definition  |
 
-- UnlockBlock
+- Block
```
Comment (thibault-martinez) [inline documentation/docs/libraries/nodejs/api_reference.md]: :trollface:

File: documentation/docs/libraries/nodejs/api_reference.md [hunk 11]
```diff
@@ -1042,13 +1042,13 @@ All fields are optional.
 | address | `string` | Output address |
 | amount  | `amount` | Output amount  |
 
-- UnlockBlockDto
+- BlockDto
```
Comment (thibault-martinez) [inline documentation/docs/libraries/nodejs/api_reference.md]: :trollface:

File: documentation/docs/libraries/nodejs/api_reference.md [hunk 12]
```diff
@@ -1025,7 +1025,7 @@ All fields are optional.
 | Field        | Type                           | Description         |
 | ------------ | ------------------------------ | ------------------- |
 | essence      | `TransactionPayloadEssenceDto` | Transaction essence |
-| unlockBlocks | `UnlockBlockDto[]`             | Unlock blocks       |
+| blocks | `BlockDto[]`             | Unlock blocks       |
```
Comment (thibault-martinez) [inline documentation/docs/libraries/nodejs/api_reference.md]: :trollface:

File: documentation/docs/libraries/nodejs/examples.mdx [hunk 13]
```diff
@@ -528,7 +528,7 @@ Example of a message with a `SignedTransaction` payload:
 				],
 				"payload": null
 			},
-			"unlockBlocks": [
+			"blocks": [
```
Comment (thibault-martinez) [inline documentation/docs/libraries/nodejs/examples.mdx]: 😡

File: documentation/docs/libraries/nodejs/examples.mdx [hunk 14]
```diff
@@ -541,17 +541,17 @@ Example of a message with a `SignedTransaction` payload:
...
-* `payload`: each `SignedTransaction`(payload type 0) can include additional payload(s) such as `TaggedPayload` (payload type 1), etc. Meaning, any value-based messages can also contain arbitrary data and its key index. It is also an example how individual payloads can be encapsulated on different levels of concern
+* `blocks`: it includes a transaction signature(s) (currently based on `Ed25519` scheme) that proofs token ownership based on a valid seed. Needless to say, only the valid seed owner is able to correctly sign the given transaction and proof the ownership of tokens under the given output(s). Each input `output` has to have a corresponding `unblockBlocks` entry in case more `outputs` are used to fund the operation either using the given signature or as a reference to the existing signature
```
Comment (thibault-martinez) [inline documentation/docs/libraries/nodejs/examples.mdx]: 😡

File: documentation/docs/libraries/python/examples.mdx [hunk 15]
```diff
@@ -565,7 +565,7 @@ Example of a message with `SignedTransaction` payload:
                   "treasury_transaction":"None"
                }
             },
-            "unlock_blocks":[
+            "blocks":[
```
Comment (thibault-martinez) [inline documentation/docs/libraries/python/examples.mdx]: 🆘

File: documentation/docs/libraries/python/examples.mdx [hunk 16]
```diff
@@ -590,12 +590,12 @@ Example of a message with `SignedTransaction` payload:
...
-* `payload`: each `SignedTransaction` can include additional payload(s) such as `TaggedPayload`, etc. Meaning, any value-based messages can also contain arbitrary data and its key index. It is also an example how individual payloads can be encapsulated on different levels of concern
+* `blocks`: it includes a transaction signature(s) (currently based on `Ed25519` scheme) that proofs token ownership based on a valid seed. Needless to say, only valid seed owner is able to correctly sign the given transaction and proofs the ownership of tokens under the given output(s). Each input `output` has to have a corresponding `unblock_block` entry in case more `outputs` are used to fund the operation either using the given signature or as a reference to existing signature
```
Comment (thibault-martinez) [inline documentation/docs/libraries/python/examples.mdx]: 🆘

File: documentation/docs/libraries/python/examples.mdx [hunk 17]
```diff
@@ -636,7 +636,7 @@ Output example:
                ],
                "payload":"None"
             },
-            "unlock_blocks":[
+            "blocks":[
```
Comment (thibault-martinez) [inline documentation/docs/libraries/python/examples.mdx]: ⚠️

File: documentation/docs/libraries/python/api_reference.md [hunk 18]
```diff
@@ -873,11 +873,11 @@ A dict with the following key/value pairs.
 ```python
 transaction = {
     'essence': RegularEssence,
-    'unlock_blocks': list[UnlockBlock]
+    'blocks': list[Block]
```
Comment (thibault-martinez) [inline documentation/docs/libraries/python/api_reference.md]: ⛔

File: documentation/docs/libraries/python/api_reference.md [hunk 19]
```diff
@@ -873,11 +873,11 @@ A dict with the following key/value pairs.
...
 }
 ```
 
-Please refer to [RegularEssence](#regularessence), and [UnlockBlock](#unlockblock) for the details of these types.
+Please refer to [RegularEssence](#regularessence), and [Block](#unlockblock) for the details of these types.
```
Comment (thibault-martinez) [inline documentation/docs/libraries/python/api_reference.md]: ⛔

File: Cargo.toml (hunk 1 above)
Comment (samuel-rufi) [reply Cargo.toml:18]: Align! :trollface:

File: README.md [hunk 20]
```diff
@@ -8,8 +8,8 @@ For value transfers we recommend using [wallet.rs](https://github.com/iotaledger
 
 This library allows you to do the following:
 
-* Create messages with indexation and transaction payloads
-* Get messages and outputs
+* Create blocks with indexation and transaction payloads
```
Comment (samuel-rufi) [inline README.md]: I know this isn't really part of your PR, but maybe change the sentence to:

"Create blocks with tagged data and transaction payloads"

File: bindings/nodejs/examples/offline_signing/example_signed_transaction.json (hunk 3 above)
Comment (samuel-rufi) [reply bindings/nodejs/examples/offline_signing/example_signed_transaction.json]: it should be `unlocks`

File: bindings/nodejs/examples/07_get_block_data.ts [hunk 21]
```diff
@@ -0,0 +1,42 @@
...
+
+    try {
+        // Create block with no payload
+        const block = await client.generateBlock();
+        console.log('Block:', block, '\n');
+
+        // Send block
+        const blockId = await client.postBlock(block);
+
+        const blockData = await client.getBlockData(blockId);
```
Comment (samuel-rufi) [inline bindings/nodejs/examples/07_get_block_data.ts]: Also not really relevant for the PR (so maybe ignore): this API should actually be `getBlock()`

File: examples/08_data_block.rs (hunk 5 above)
Comment (Thoralf-M) [reply examples/08_data_block.rs:26]: Currently it doesn't work anyways, we need to update when we have a public explorer for stardust

**Review by thibault-martinez — APPROVED** [review]

**Review by samuel-rufi — APPROVED** [review]

**Pull request merged**